File: src/015/p015.erl
-module(p015).
-export([solve/0]).

factorial(0) -> 1;
factorial(N) when N > 0 -> N * factorial(N - 1).

combination(N, K) -> 
  C = factorial(N) / (factorial(K) * factorial(N - K)),
  round(C).

count_routes(K) -> combination(2 * K, K).

solve() -> count_routes(20).

File: src/015/p015.exs
defmodule Problem015 do
  defp factorial(0), do: 1
  defp factorial(n) when n > 0, do: n * factorial(n - 1)

  defp combination(n, k) do
    combination = factorial(n) / (factorial(k) * factorial(n - k))
    round(combination)
  end

  defp count_routes(k), do: combination(2 * k, k)

  def solve, do: count_routes(20)
end

IO.puts Problem015.solve
